core/store/src/trie/split.rs

```diff
@@ -692,6 +743,51 @@ mod tests {
             assert!(trie_descent.next_step().is_some());
         }
 
+        #[test]
+        #[should_panic]
+        fn key_too_short() {
+            let mut arena = STArena::new("test".to_string());
+
+            // A single entry with 1-nibble key (not long enough for an account ID)
+            let mut subtrees = [new_leaf(&mut arena, &[], &[]); SUBTREES.len()];
+            subtrees[0] = new_leaf(&mut arena, &[1], &[1u8; 100]);
+            init(&mut arena, subtrees).find_mem_usage_split();
```

Comment:
I don't think this code should panic in production, can you return an error? In the test you can check that error was returned, in prod you'd log it and proceed without resharding I guess?

Reply:
This panic means that we have a key that's not a valid account ID in the accounts subtree. So it seems like a corrupt storage. And if a node continues to operate with a corrupt storage, bad things might happen. I'm no sure if we have some clear rules when to panic, but to me it seemed appropriate here.